src/main/scala/components/memory/memory-port-bus-io.scala

```diff
@@ -0,0 +1,35 @@
+// IO Bundles for the bus that connecting the memory ports to the backing memory.
+
+package dinocpu
+
+import chisel3._
+import chisel3.util._
+
+// A Bundle used for representing a memory access by instruction memory or data memory.
+class Request extends Bundle {
+  val address      = UInt(32.W)
+  val writedata    = UInt(32.W)
+  val operation    = UInt(1.W)
```

Comment:
I think this needs to be two bits.

Is there a way to type it as a MemoryOperation since you have the enum? There should be a way to infer the width (in case we add another operation type).